x/programs/runtime/runtime.go

```diff
@@ -65,6 +65,23 @@ func (r *WasmRuntime) AddProgram(program codec.Address, bytes []byte) (*Program,
 }
 
 func (r *WasmRuntime) CallProgram(ctx context.Context, callInfo *CallInfo) ([]byte, error) {
+	result, err := r.callProgram(ctx, callInfo)
+	if err != nil {
+		trap := convertToTrap(err)
+		// https://github.com/bytecodealliance/wasmtime-go/blob/v14.0.0/trap.go#L30
+		switch *trap.Code() {
+		case 9:
+			return append(Err, CallPanicked), err
+		case 11:
+			return append(Err, OutOfFuel), err
+		default:
+			return append(Err, ExecutionFailure), err
+		}
+	}
```

Comment:
Don't we already have a single source of truth for converting trap-errors into our own set of errors are whatever? 

There should be an abstraction here and there definitely shouldn't be random numbers just floating in the code. At the very least, these numbers should be defined as constants.

Reply:
Thanks, yeah I just used the constants at this link in wasmtime, it's not going to break if we update the dependency that way.